### PR TITLE
[migration templates fix] migration generator template, resource_owner integer to references

### PR DESCRIPTION
**Pull request by atomiyama**
### Summary

In over Rails5.1, Default primary key has been bigint.
If Rails version is below 5.1, resource_owner_id column will be set INT, above 5.1 it will be set BIGINT

### CODE REVIEW

File: lib/generators/doorkeeper/templates/migration.rb.erb [hunk 1]
```diff
@@ -10,9 +10,9 @@ class CreateDoorkeeperTables < ActiveRecord::Migration<%= migration_version %>
       t.timestamps             null: false
     end
 
-    add_index :oauth_applications, :uid, unique: true
+    add_index :oauth_applications, :uid, unique: true, <%= primary_id_option %>
```
Comment (nbulaj) [inline lib/generators/doorkeeper/templates/migration.rb.erb]: Why do we need to add this option for `add_index` ?

File: lib/generators/doorkeeper/templates/migration.rb.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class CreateDoorkeeperTables < ActiveRecord::Migration<%= migration_version %>
   def change
-    create_table :oauth_applications do |t|
+    create_table :oauth_applications, <%= primary_id_option %> do |t|
```
Comment (nbulaj) [inline lib/generators/doorkeeper/templates/migration.rb.erb]: Could not find in any of my Rails 5.2 projects this option, can you please share a link to PR or any article describing this?

Just run `rails g migration CreateUsers name:string` and got:

```ruby
class CreateUsers < ActiveRecord::Migration[5.2]
  def change
    create_table :users do |t|
      t.string :name
    end
  end
end
```

File: lib/generators/doorkeeper/templates/migration.rb.erb [hunk 3]
```diff
@@ -10,9 +10,9 @@ class CreateDoorkeeperTables < ActiveRecord::Migration<%= migration_version %>
       t.timestamps             null: false
     end
 
-    add_index :oauth_applications, :uid, unique: true
+    add_index :oauth_applications, :uid, unique: true, <%= primary_id_option %>
 
-    create_table :oauth_access_grants do |t|
+    create_table :oauth_access_grants, <%= primary_id_option %> do |t|
```
Comment (nbulaj) [inline lib/generators/doorkeeper/templates/migration.rb.erb]: I think `,` must be included in the template only if primary ID option is not null...

**Comment by nbulaj** [comment]
Hi @atomiyama . As far as I know Rails generates bigint by default. And [this article](http://www.mccartie.com/2016/12/05/rails-5.1.html) just approves my thoughts. Fix me if I wrong..

File: lib/generators/doorkeeper/templates/migration.rb.erb (hunk 1 above)
Comment (atomiyama) [reply lib/generators/doorkeeper/templates/migration.rb.erb]: it is not needed, will fixed.

File: lib/generators/doorkeeper/templates/migration.rb.erb (hunk 3 above)
Comment (atomiyama) [reply lib/generators/doorkeeper/templates/migration.rb.erb]: included

**Comment by atomiyama** [comment]
At rails 5.2, if you want to use User table as the resource owner and you add foreign keys to your migration, your "db:migrate" cause data type unmatch error, because User table's primary key would be bigint by default.

**Comment by nbulaj** [comment]
@atomiyama is it something specific for MySQL or how? Never faced this problem.

Also not so much results found with this use case https://github.com/search?q=ActiveRecord%3A%3AMigration+create_table+id%3A+%3Abigint+language%3Aruby&type=Code

**Comment by atomiyama** [comment]
create_table is not needed to be passed id option. it has been deleted.
I faced this migration problem data type unmatch in Rails5.2 w/ mysql

**Comment by nbulaj** [comment]
Oh, now I see it from you last changes :) Makes sense

But I prefer to use `t.belongs_to :resource_owner` (or `t.references :resource_owner`) because it will automatically use bigint for Rails >= 5.1 and `integer` for previous releases

**Comment by atomiyama** [comment]
it makes sense! this PR should be close. thank you for your reviews :)

**Comment by nbulaj** [comment]
You @atomiyama can produce a new PR (or fix this one) to use `t.belongs_to` instead of plain `t.integer`, it would be great! Thanks

**Comment by atomiyama** [comment]
@nbulaj thanks for your kind. it has been fixed.

**Comment by atomiyama** [comment]
@nbulaj done! :)

**Pull request merged**